--- a/src/FilterSelector/FiltersView/FiltersView.cpp
+++ b/src/FilterSelector/FiltersView/FiltersView.cpp
@@ -431,11 +431,15 @@
   }
   onItemClicked(index);
   if (item->isFave()) {
-    _faveContextMenu->deleteLater();
+    if (_faveContextMenu) {
+      _faveContextMenu->deleteLater();
+    }
     _faveContextMenu = itemContextMenu(MenuType::Fave, item);
     _faveContextMenu->exec(ui->treeView->mapToGlobal(point));
   } else {
-    _filterContextMenu->deleteLater();
+    if (_filterContextMenu) {
+      _filterContextMenu->deleteLater();
+    }
     _filterContextMenu = itemContextMenu(MenuType::Filter, item);
     _filterContextMenu->exec(ui->treeView->mapToGlobal(point));
   }
